### ISSUE TITLE
When i rotate myself the bubble tremble

### ISSUE THREAD

**Issue by Lanselotto** [Haplo064/ChatBubbles#3]
Chat bubble tremble a lot when i rotate my body with standard camera, either with A/D or with the right mouse button. I think it's because the arrow of the chat bubble keep switching frome left to right side. It would be muche better if the chat bubbles where static

**Comment by Haplo064** [comment]
Yeah, you're correct in that it's switching from left to right. It's dependant on where they player character is in the main view.
I've got ideas to find out what's causing it and lock it down, but it's not a big priority at the moment.
This'll stay open until I resolve it.

**Comment by Lanselotto** [comment]
it might be a good workaround to let me turn off my character bubble

**Comment by Omega2k3** [comment]
I get this whenever I move around at all, and it's not just a minor tremble, it shakes pretty rapidly left and right a good distance.

**Comment by Lanselotto** [comment]
Yeah, that's why i made a workaround but seems haplo forgot to release it

**Comment by guyute628** [comment]
I am also getting this issue :( super sad as this is a feature that I’ve always wanted <3 any ideas what could be causing it? Anything I could try?

**Comment by Lanselotto** [comment]
yes, the renderer of the game change the bubble position based on yours pos and orientation so when walk it tremble. You have to fuck up with a lot of retro engigneering to fix this. Just download and compile my fork so at least you can hide your bubble

**Comment by jamesusa19** [comment]
@Lanselotto, sorry not familiar with the plugin files. If possible, could you describe how to make the necessary changes?

**Comment by Lanselotto** [comment]
I have released an unofficial version, with an hide self bubble button, just download and unzip it in %AppData%\XIVLauncher\installedPlugins . Keep in mind that my version is unofficial so don't blame Haplo if something doesn't work. I don't take any responsibility anyway, use it at your own risk.

**Comment by Haplo064** [comment]
I wish I had better news, but all my attempts so far to fix this have failed.
It is high on my own internal issue list, as it annoys the shit out of me as well!

**Comment by vinajuliette** [comment]
I've recently noticed I can see this effect on nameplates sometimes too, though it's more subtle

**Comment by orangejuiceXIV** [comment]
So I'm not an expert in 3D graphics/rendering, but I do work alot with 2D transformations, and have an solid understanding of the math behind affine transformations, perspective transforms, etc.

Isn't the jitter simply the fact that the the way the function deciding where to draw the text box isn't completely continuous/smooth? Essentially it should the location of where to draw the center of the chat box should be should be computed with some continuous/smooth function on the the players position relative to the camera position/angle. Since the function is continuous/smooth, there should not be any jitter and the motion of the bubbles should be entirely smooth with respect to motion of the camera and the player. Any sort of functional mapping with discontinuities/lack of smoothness fundamentally cannot be expected to behave nicely.

**Comment by muesli-mage** [comment]
ffxiv_dx11.exe+133F041 sub dx, cx
change to xor dx, dx

**Comment by MKhayle** [comment]
> ffxiv_dx11.exe+133F041 sub dx, cx change to xor dx, dx

I'll try to see about this at some time, ty

**Comment by Lovasz-Akos** [comment]
+1 :)
